core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -568,6 +568,56 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     Stream.eval(fstream)
   }
 
+  def conflate[F2[x] >: F[x], O2 >: O](implicit
```

Comment:
I don't love this signature. Feels very adhoc.

Reply:
I was thinking on `Channel`-based implementation that you suggested on Reddit and come up with some method signatures that I think would feel better:

1. `def conflateChunks(implicit F: Concurrent[F]): Stream[F, Chunk[O]]` -- the base method for all others.
     As `Channel.stream` emits all collected data in a singular chunk, it feels natural to give a user
     access to them w/o any additional operations on elements.
     Also, I think it's worth doing even for current implementation, as pulling chunks is cheaper
     than pulling elements and chunk concat is O(1).
2. `def conflate[O2](zero: O2)(f: (O2, O) => O2)(implicit F: Concurrent[F]): Stream[F, O2]` -- I think
    `conflate` series of methods  should be similar in signatures to `fold/scan`.
3. `def conflate1(f: (O, O) => O)(implicit F: Concurrent[F]): Stream[F, O]` - again, there are `fold1/scan1`
    and we just 'swapped' Akka's `conflate/conflateWithSeed`.
4. `def conflateSemigroup(implicit F: Concurrent[F], O: Semigroup[O]): Stream[F, O]` -- once more
    we do like `foldMonoid/scanMonoid` except that we don't need `empty`;
5. `def conflateMap` -- I'm not sure about that one, but why not?

On a side note, Cambridge Dictionary suggest 'combine, fuse, meld, merge' as synonims. 'Merge' is taken, 'fuse' and 'meld' are meh, but 'combine' is a-ok especially because we use `Semigroup`. On the other hand `conflate` would be more weel known for the people with Akka/Reactive Streams experience, much like `switchMap`.

Reply:
If we do something like `conflateChunks`, we're getting very close to the behavior of `prefetchN` -- which might be a sign that we should just offer a `prefetchN` variant that combines all chunks when dequeuing. Using `conflateWithSeed` as the base operation allows the conflation to drop / summarize data instead of storing each element. On the other hand, `conflate` suffers from loss of backpressure as-is -- there's no limit to how much is pulled from source stream, so if downstream isn'e expedient, we could run out of memory. `prefetchN` doesn't have this issue as it uses a bounded channel to transfer elements.

Reply:
On second thought, `prefetchN` already does the right thing - accumulates chunks that arrive while downstream is processing. So `conflate` is as simple as:

```scala
  def conflate[F2[x] >: F[x], O2 >: O](implicit
      F: Concurrent[F2],
      O: Semigroup[O2]
  ): Stream[F2, O2] =
    prefetchN(Int.MaxValue).chunks.map(_.combineAll)
```

Reply:
I pushed some new implementations using your suggested signatures plus the addition of a `chunkLimit` param on each. Please take a look.

While channel does the right thing, `prefetchN` doesn't as it maintains the source chunk structure. The version of `conflateChunks` I just pushed uses a `Channel` directly and inlines the chunk conflation logic using `unconsFlatMap`.

Reply:
I like it very much!

One nitpick -- I'm not sure about `chunkLimit`, maybe it's easier to use `Channel.unbounded`? We still have no control over number of elements in each conflated chunk, so it's not like very have any control over consumed memory.

Reply:
I'd prefer to force folks to make a decision to use `Int.MaxValue` here instead of, by default, allowing unbounded memory usage. It's a bit of a speed bump to get folks to think about memory usage. Not sure `chunkLimit` is the right name given it's really a limit on number of chunks pulled from source stream and the max size of chunks emitted downstream.